Tests/ML/histopathology/utils/test_metrics_utils.py

```diff
@@ -64,27 +72,40 @@ def assert_equal_lists(pred: List, expected: List) -> None:
-                               select=('lowest_pred', 'lowest_att'))
-    assert_equal_lists(top_fn, [(3, 0.4, [1, 2], [Tensor([0.25]), Tensor([0.23])]),
-                                (2, 0.7, [2, 3], [Tensor([0.18]), Tensor([0.15])])])
-    assert_equal_lists(bottom_fn, [(3, 0.4, [3, 4], [Tensor([0.20]), Tensor([0.21])]),
-                                   (2, 0.7, [1, 4], [Tensor([0.10]), Tensor([0.13])])])
-
-    top_tp = select_k_tiles(test_dict, n_slides=nslides, label=1, n_tiles=ntiles,
-                            select=('highest_pred', 'highest_att'))
-    bottom_tp = select_k_tiles(test_dict, n_slides=nslides, label=1, n_tiles=ntiles,
-                               select=('highest_pred', 'lowest_att'))
-    assert_equal_lists(top_tp, [(4, 1.0, [3, 4], [Tensor([0.37]), Tensor([0.35])]),
-                                (2, 0.7, [2, 3], [Tensor([0.18]), Tensor([0.15])])])
-    assert_equal_lists(bottom_tp, [(4, 1.0, [2, 1], [Tensor([0.31]), Tensor([0.33])]),
-                                   (2, 0.7, [1, 4], [Tensor([0.10]), Tensor([0.13])])])
+    bottom_fn = select_k_tiles(test_dict, n_slides=nslides, label=1, n_tiles=ntiles, select=('lowest_pred', 'lowest_att'))
+    assert_equal_lists(top_fn, [(5, Tensor([0.7, 0.3]), [1, 4], [Tensor([0.43]), Tensor([0.25])]),
+                                (3, Tensor([0.6, 0.4]), [1, 2], [Tensor([0.25]), Tensor([0.23])])])
+    assert_equal_lists(bottom_fn, [(5, Tensor([0.7, 0.3]), [2, 3], [Tensor([0.01]), Tensor([0.07])]),
+                                   (3, Tensor([0.6, 0.4]), [3, 4], [Tensor([0.20]), Tensor([0.21])])])
+
+    # TN
+    top_tn = select_k_tiles(test_dict, n_slides=nslides, label=0, n_tiles=ntiles, select=('highest_pred', 'highest_att'))
+    bottom_tn = select_k_tiles(test_dict, n_slides=nslides, label=0, n_tiles=ntiles, select=('highest_pred', 'lowest_att'))
+    assert_equal_lists(top_tn, [(6, Tensor([0.8, 0.2]), [4, 1], [Tensor([0.55]), Tensor([0.53])]),
+                                (1, Tensor([0.6, 0.4]), [3, 4], [Tensor([0.2]), Tensor([0.15])])])
+    assert_equal_lists(bottom_tn, [(6, Tensor([0.8, 0.2]), [2, 3], [Tensor([0.11]), Tensor([0.17])]),
+                                  (1, Tensor([0.6, 0.4]), [2, 1], [Tensor([0.00]), Tensor([0.10])])])
+
+    # FP
+    top_tn = select_k_tiles(test_dict, n_slides=nslides, label=0, n_tiles=ntiles, select=('lowest_pred', 'highest_att'))
+    bottom_tn = select_k_tiles(test_dict, n_slides=nslides, label=0, n_tiles=ntiles, select=('lowest_pred', 'lowest_att'))
+    assert_equal_lists(top_tn, [(8, Tensor([0.01, 0.99]), [1, 3], [Tensor([0.73]), Tensor([0.37])]),
+                                (7, Tensor([0.1, 0.9]), [1, 3], [Tensor([0.63]), Tensor([0.27])])])
+    assert_equal_lists(bottom_tn, [(8, Tensor([0.01, 0.99]), [4, 2], [Tensor([0.15]), Tensor([0.31])]),
+                                  (7, Tensor([0.1, 0.9]), [4, 2], [Tensor([0.05]), Tensor([0.21])])])
```

Comment:
These tests make sense, but I worry there's a risk of the expected values written here and the originals in `test_dict` going out of sync in later changes to this file. Should we create a ticket to later refactor this to access the corresponding values from `test_dict`?

Reply:
Thanks. I created a ticket for this